File: packages/jetbrains-plugin/src/test/kotlin/com/evalops/composer/toolwindow/RuntimeStatusTest.kt
package com.evalops.composer.toolwindow

import com.evalops.composer.api.AgentEvent
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Test

class RuntimeStatusTest {

    @Test
    fun `formats compacting status events`() {
        val event = AgentEvent.Status(status = "compacting")
        assertEquals("Compacting conversation...", formatRuntimeStatus(event))
    }

    @Test
    fun `trims status values before formatting`() {
        val event = AgentEvent.Status(status = " compacting ")
        assertEquals("Compacting conversation...", formatRuntimeStatus(event))
    }

    @Test
    fun `formats compaction events`() {
        val event = AgentEvent.Compaction(
            summary = "Summary",
            firstKeptEntryIndex = 4,
            tokensBefore = 1200,
            auto = true,
            timestamp = "2026-04-01T00:00:00Z"
        )
        assertEquals("Compacted conversation automatically", formatRuntimeStatus(event))
    }

    @Test
    fun `ignores unrelated events`() {
        assertNull(formatRuntimeStatus(AgentEvent.AgentStart()))
    }
}

File: packages/vscode-extension/src/lib/runtime-status.test.ts
import { describe, expect, it } from "vitest";
import { formatVscodeRuntimeStatus } from "./runtime-status.js";

describe("vscode runtime status", () => {
	it("formats compacting status events", () => {
		expect(
			formatVscodeRuntimeStatus({
				type: "status",
				status: "compacting",
				details: {},
			}),
		).toBe("Compacting conversation...");
	});

	it("formats generic status events", () => {
		expect(
			formatVscodeRuntimeStatus({
				type: "status",
				status: "planning",
				details: {},
			}),
		).toBe("Status: planning");
	});

	it("formats live tool summaries without the generic prefix", () => {
		expect(
			formatVscodeRuntimeStatus({
				type: "status",
				status: "Read app.ts",
				details: { kind: "tool_execution_summary" },
			}),
		).toBe("Read app.ts");
	});

	it("trims status values before formatting", () => {
		expect(
			formatVscodeRuntimeStatus({
				type: "status",
				status: " compacting ",
				details: {},
			}),
		).toBe("Compacting conversation...");
	});

	it("formats compaction events", () => {
		expect(
			formatVscodeRuntimeStatus({
				type: "compaction",
				summary: "Summary",
				firstKeptEntryIndex: 3,
				tokensBefore: 1000,
				auto: true,
				timestamp: new Date(0).toISOString(),
			}),
		).toBe("Compacted conversation automatically");
	});
});
